app/models/firm.rb

```diff
@@ -109,7 +110,7 @@ def longitude=(value)
   def geocode!(latitude = nil, longitude = nil)
     self.latitude = latitude
     self.longitude = longitude
-    save!(callbacks: false)
+    save!(callbacks: false, validate: false)
```

Comment:
When you're not validating `save!` is unnecessary. It would never raise because the validations aren't being run.

I would favour us knowing before we get into this branch that we're already valid so we don't have to disable validation. That would make more sense, and be more robust.

My suggestion is that you revert this stuff, and just short-circuit the guard around creating the geocoding job with `valid?` as the first operand.